include/sparrow/layout.hpp

```diff
@@ -108,8 +111,10 @@ namespace sparrow
         using const_iterator = primitive_layout_iterator<T, true>;
 
         size_type size() const;
-        reference element(size_type i);
-        const_reference element(size_type i) const;
+        inner_reference value(size_type i);
+        inner_const_reference value(size_type i) const;
+
+        bool has_value(size_type i) const;
```

Comment:
I wonder if the `value` / `has_value` methods should be private, since they are meant to be used by the `reference_proxy` and the iterator classes only. @Klaim do you have any thought on this?

Reply:
So this means we cannot test the iterator anymore?

Reply:
You can, through the reference proxy:

```cpp
 CHECK_EQ(*iter++, lt[i].value());